src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,352 @@
+        extension_function_dict["input_sql_types"] = extension_function_properties.input_sql_types
+        extension_function_dict["return_sql"] = extension_function_properties.return_sql
+        extension_function_dict["runtime_version"] = extension_function_properties.runtime_version
+        extension_function_dict["all_imports"] = extension_function_properties.all_imports
+        extension_function_dict["all_packages"] = extension_function_properties.all_packages
+        extension_function_dict["handler"] = extension_function_properties.handler
+        extension_function_dict["external_access_integrations"] = extension_function_properties.external_access_integrations
+        extension_function_dict["secrets"] = extension_function_properties.secrets
+        extension_function_dict["inline_python_code"] = extension_function_properties.inline_python_code
+        extension_function_dict["native_app_params"] = extension_function_properties.native_app_params
+        extension_function_dict["raw_imports"] = extension_function_properties.raw_imports
+        extension_function_dict["replace"] = extension_function_properties.replace
+        extension_function_dict["if_not_exists"] = extension_function_properties.if_not_exists
+        extension_function_dict["execute_as"] = extension_function_properties.execute_as
+        extension_function_dict["anonymous"] = extension_function_properties.anonymous
+        raw_func = extension_function_properties.func
+        extension_function_dict["func"] = raw_func.__name__ if isinstance(raw_func, Callable) else raw_func
+        return extension_function_dict
+
+
+    def __SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_callback_append_to_list(callback_return_list, extension_function_properties):
+        extension_function_dict = __SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_transform_snowpark_object_to_json(extension_function_properties)
+        #TODO: print(extension_function_dict) here in case of downstream UDF creation failures
+        callback_return_list.append(extension_function_dict)
+        return False
+
+    return functools.partial(__SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_callback_append_to_list, __SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_callback_return_list)
+
+with open({py_file}, mode='r', encoding='utf-8') as udf_code:
+    code = udf_code.read()
+
+
+snowflake.snowpark.context._is_execution_environment_sandboxed_for_client = True
+snowflake.snowpark.context._should_continue_registration = __SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_callback_replacement()
+snowflake.snowpark.session._is_execution_environment_sandboxed_for_client = True
+
+# Remove reference to the callback from globals() since the function is already assigned to snowflake.snowpark.context._should_continue_registration
+del __SNOWFLAKE_CLI_NATIVE_APP_INTERNAL_callback_replacement
```

Comment:
nit: that's a bit odd, I would have done something like:

```
g = globals()
del g['<internal name>']
exec(code, g)
```

Reply:
True, if we want to delete just they key from the globals() dict. But in this case, since the snowpark callback has already been assigned the callable, I deleted the reference to the callable. Not strongly opinionated, so can go either way.